src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/updateotherparty/UpdateOtherPartyAboutToSubmitHandlerTest.java

```diff
@@ -440,6 +441,168 @@ public void givenSscs1CaseOtherPartyWantsToAttendYes_thenHearingTypeNotChangedAn
         assertEquals(hearingType, response.getData().getAppeal().getHearingType());
     }
 
+    @Test
+    public void givenAnyCaseUnavailableDatesSelectedButEmptyExcludedDates_thenThrowError() {
+        CcdValue<OtherParty> otherParty1 = buildOtherParty("Yes", NO);
+        CcdValue<OtherParty> otherParty2 = buildOtherParty("Yes", NO);
+        CcdValue<OtherParty> otherParty3 = buildOtherParty("Yes", NO);
+
+        otherParty1.getValue().getHearingOptions().setScheduleHearing("Yes");
+        otherParty2.getValue().getHearingOptions().setScheduleHearing("Yes");
+        otherParty3.getValue().getHearingOptions().setScheduleHearing("Yes");
+
+        otherParty1.getValue().setRole(Role.builder().name("ReceivingParent").build());
+        otherParty2.getValue().setRole(Role.builder().name("PayingParent").build());
+        otherParty3.getValue().setRole(Role.builder().name("Other").build());
+
+        Appeal appeal = Appeal.builder()
+            .benefitType(BenefitType.builder().code(Benefit.CHILD_SUPPORT.getShortName()).build())
+            .build();
+
+        SscsCaseData sscsCaseData = SscsCaseData.builder()
+            .appeal(appeal)
+            .otherParties(Arrays.asList(otherParty1, otherParty2, otherParty3)).build();
+
+        when(caseDetails.getCaseData()).thenReturn(sscsCaseData);
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertEquals(0, response.getWarnings().size());
+        assertEquals(2, response.getErrors().size());
+
+        Iterator<String> iterator = response.getErrors().iterator();
+        String error1 = iterator.next();
+        String error2 = iterator.next();
+        assertEquals("Add a start date for unavailable dates", error1);
+        assertEquals("Add an end date for unavailable dates", error2);
+    }
+
+    @Test
+    public void givenAnyCaseUnavailableDatesSelectedButNoDatesProvided_thenThrowError() {
+        CcdValue<OtherParty> otherParty = buildOtherParty("Yes", NO);
+        otherParty.getValue().getHearingOptions().setScheduleHearing("Yes");
+        otherParty.getValue().setRole(Role.builder().name("ReceivingParent").build());
+
+        ExcludeDate excludeDate1 = ExcludeDate.builder().value(DateRange.builder().start("").end("").build()).build();
+        ExcludeDate excludeDate2 = ExcludeDate.builder().value(DateRange.builder().start(null).end(null).build()).build();
+        ExcludeDate excludeDate3 = ExcludeDate.builder().value(DateRange.builder().start("2023-01-01").end("2023-01-02").build()).build();
+
+        otherParty.getValue().getHearingOptions().setExcludeDates(List.of(excludeDate1, excludeDate2, excludeDate3));
+
+        Appeal appeal = Appeal.builder()
+            .benefitType(BenefitType.builder().code(Benefit.CHILD_SUPPORT.getShortName()).build())
```

Comment:
BenefitType.builder().code(Benefit.CHILD_SUPPORT.getShortName()).build() as separate var

Reply:
Check for simillar stuff below